pkgs/servers/sql/postgresql/generic.nix

```diff
@@ -341,21 +340,17 @@ let
 
       postPatch = ''
         substituteInPlace "src/Makefile.global.in" --subst-var out
-        # Hardcode the path to pgxs so pg_config returns the path in $dev
-        substituteInPlace "src/common/config_info.c" --subst-var dev
+        cat ${./pg_config.env.mk} >> src/common/Makefile
       '';
 
       postInstall =
         ''
           moveToOutput "bin/ecpg" "$dev"
           moveToOutput "lib/pgxs" "$dev"
 
-          # Pretend pg_config is located in $out/bin to return correct paths, but
-          # actually have it in -dev to avoid pulling in all other outputs. See the
-          # pg_config.sh script's comments for details.
-          moveToOutput "bin/pg_config" "$dev"
-          install -c -m 755 "${pg_config}"/bin/pg_config "$out/bin/pg_config"
-          wrapProgram "$dev/bin/pg_config" --argv0 "$out/bin/pg_config"
+          rm "$out/bin/pg_config"
+          make -C src/common pg_config.env
+          install -D src/common/pg_config.env "$dev/nix-support/pg_config.env"
```

Comment:
While I don't expect much to change around pg_config: how will we make sure that there are no larger changes, e.g. more vars exposed?
Personally, I'd like to something like `pg_config` being built into e.g. another output (or some dir in $dev) and a test making sure that there are no new vars.

Reply:
> We can ofc only test this for native builds

Should've been more explicit here, but you're right :) 

> I'll think about it and come up with something.

Cool, thanks!